cla_backend/apps/cla_eventlog/management/commands/find_and_delete_old_cases.py

```diff
@@ -0,0 +1,28 @@
+from django.core.management.base import BaseCommand
+from dateutil.relativedelta import relativedelta
+
+from legalaid.models import Case
+from cla_butler.tasks import DeleteOldData
+
+
+class FindAndDeleteCasesUsingCreationTime(DeleteOldData):
+    def get_eligible_cases(self):
+        super(FindAndDeleteCasesUsingCreationTime, self)._setup()
+        two_years = self.now - relativedelta(years=2)
+
+        return Case.objects.filter(created__lte=two_years).exclude(log__created__gte=two_years)
+
+
+class Command(BaseCommand):
+    help = (
+        "Find or delete cases that are 2 years old or over that were not deleted prior to the task command being fixed"
+    )
+
+    def handle(self, *args, **kwargs):
+        if args[0] == "find":
+            cases = FindAndDeleteCasesUsingCreationTime().get_eligible_cases()
+            print(cases)
```

Comment:
Do you really want to print thousands of cases to the stdout?

Reply:
Yeah true. Forgot we have like 100k+ cases to go through

I remember you talking about transactions in the database. Will that act as a cache, so we can restore/rollback the cases if anything goes wrong? (I had this in mind, but didn't know how to achieve it).

Reply:
I think you would need to explicitly invoke a transaction. I recommend searching for "atomic" in the Django docs. Make sure to be on the right Django version, though: that stuff will have evolved various helpers that we won't have access to on our old version

Reply:
Right I checked it out and every transaction is committed to the database by default. So we can just undo the transaction from the database. Awesome

Reply:
Can you link me to what you're looking at, please?

Reply:
https://django.readthedocs.io/en/1.7.x/topics/db/transactions.html#django-s-default-transaction-behavior

Reply:
Right, so yes transactions are automatically committed when they complete.
And every query is wrapped in a transaction if it's not explicitly part of a bigger transaction.
So if you execute a query, it will run inside a transaction and either be committed or rolled back, and then the transaction is done and gone.

You can also put multiple queries into one transaction, so that they are either all committed or all rolled back together. But you have to actively use something like `atomic` to make these queries share a transaction, and I don't think we are doing that here.

And, either way, I think "So we can just undo the transaction from the database" suggests you might be misunderstanding this a bit. Once a transaction is committed, it's committed. You can roll back a transaction from within it, *instead* of committing it, so we could use one to avoid repeating what happened before, where the `Log`s were deleted but the `Case` failed to delete and we were left with a `Case` with no `Log`s, by wrapping those two deletions in a transaction, but we couldn't use it to restore data after the fact, in case that's what you were thinking.

Reply:
Oh ok. So if we wanted to e.g. restore deleted cases can we rollback the database? I was mostly concerned with what we will do in that kinda situation

Reply:
Not particularly easily, no. We just need to be confident in what we're deleting

Reply:
Cool thanks for clearing that up 👍